library/common/engine.cc

```diff
@@ -94,14 +95,10 @@ Engine::~Engine() {
 }
 
 void Engine::recordCounter(std::string elements, uint64_t count) {
```

Comment:
it's weird to have a scalar string called `elements`, but in any case I suggest passing it by const ref, or better yet, by string_view.

But IMO the higher level issue is that if you define your API to inc a counter based on a string, you are going to have to pay the cost of map lookups and/or locks. Is it not possible for you to have the Engine API offer a wrapper to create a handle that the caller can store, to quickly increment a counter?

Reply:
I think this is ideally what we'd like to do, but we were going for something quick to iterate on here, and lacked the familiarity with the stats API to figure out how best to accomplish this. Would welcome your perspective on a sounder approach.

Reply:
```suggestion
void Engine::recordCounter(const std::string& elements, uint64_t count) {
```

per @jmarantz.